index.js

```diff
@@ -1,21 +1,66 @@
 // package imports
 const express = require('express');
+const session = require('express-session');
+const passport = require('passport');
+const cookieParser = require('cookie-parser');
+const bodyParser = require('body-parser');
+const cors = require('cors');
+const env = require('dotenv').config();
+
 // const imports
+const dbConnect = require('./config/db.connect');
+const passportLocalConfigure = require('./config/passport/passport.local.config');
+const passportGoogleConfig = require('./config/passport/passport.google.config');
+const passportGithubConfig = require('./config/passport/passport.github.config');
+const constants = require('./constants/constants');
+
+// Routes import
 const practiceRouter = require('./modules/practice/practice.route');
+const authRouter = require('./modules/auth/auth.route');
+
+// Logger imports
+const logger = require('./config/logger/logger');
 
 // creating instance
 const app = express();
 
-// Adding middleware
-app.use(express.json());
+// Connecting database
+dbConnect();
+
+// Adding middlewares
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: true}));
+app.use((req, res, next) => {
+  res.setHeader('Access-Control-Allow-Origin', process.env.CLIENT_URL);
+  res.setHeader('Access-Control-Allow-Credentials', 'true');
+  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE');
+  res.setHeader('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept');
+  next();
+});
+app.use(
+    session({
+      secret: process.env.SESSION_CODE,
+      resave: true,
+      saveUninitialized: true,
+    }),
+);
+app.use(cookieParser(process.env.SESSION_CODE));
+app.use(passport.initialize());
+app.use(passport.session());
+passportLocalConfigure(passport);
+passportGoogleConfig(passport);
+passportGithubConfig(passport);
+
+// Configuring Routes
 app.use('/api/practice/', practiceRouter);
+app.use('/api/auth/', authRouter);
 
```

Comment:
These routes should be defined in the routes file as discussed earlier.

Reply:
these are not the routes being defined here. I ma just integrating created routes in app through app.use()